src/components/QueryForm.tsx

```diff
@@ -1,179 +1,97 @@
-  const [maxAge, setMaxAge] = useState<string | null>(null);
-  const [sex, setSex] = useState<FieldInput>(null);
-  const [diagnosis, setDiagnosis] = useState<FieldInput>(null);
-  const [isControl, setIsControl] = useState<boolean>(false);
-  const [minNumSessions, setMinNumSessions] = useState<string | null>(null);
-  const [assessmentTool, setAssessmentTool] = useState<FieldInput>(null);
-  const [imagingModality, setImagingModality] = useState<FieldInput>(null);
-  const [searchParams, setSearchParams] = useSearchParams();
+  const [minAgeError, setMinAgeError] = useState(false);
+  const [minAgeHelperText, setMinAgeHelperText] = useState('');
+  const [maxAgeError, setMaxAgeError] = useState(false);
+  const [maxAgeHelperText, setMaxAgeHelperText] = useState('');
+  const [minNumSessionsError, setMinNumSessionsError] = useState(false);
+  const [minNumSessionsHelperText, setMinNumSessionsHelperText] = useState('');
 
-  useEffect(() => {
-    if (nodeOptions.length > 1) {
-      const searchParamNodes: string[] = searchParams.getAll('node');
-      if (searchParamNodes) {
-        const matchedOptions: FieldInputOption[] = searchParamNodes
-          .map((label) => {
-            const foundOption = nodeOptions.find((option) => option.NodeName === label);
-            return foundOption ? { label, id: foundOption.ApiURL } : { label, id: '' };
-          })
-          .filter((option) => option.id !== '');
-        // If there is no node in the search params, set it to All
-        if (matchedOptions.length === 0) {
-          setSearchParams({ node: ['All'] });
-          setNode([{ label: 'All', id: 'allNodes' }]);
-        }
-        // If there is any node besides All selected, remove All from the list
-        else if (
-          matchedOptions.length > 1 &&
-          matchedOptions.some((option) => option.id === 'allNodes')
-        ) {
-          const filteredNode: FieldInputOption[] = matchedOptions.filter(
-            (n) => n.id !== 'allNodes'
-          );
-          setNode(filteredNode);
-          setSearchParams({ node: filteredNode.map((n) => n.label) });
-        } else {
-          setNode(matchedOptions);
-        }
-      }
-    }
-  }, [searchParams, setSearchParams, nodeOptions]);
-
-  function updateCategoricalQueryParams(fieldLabel: string, value: FieldInput) {
-    switch (fieldLabel) {
-      case 'Neurobagel graph':
-        setNode(value);
-        if (Array.isArray(value)) {
-          setSearchParams({ node: value.map((n) => n.label) });
-        }
-        break;
-      case 'Sex':
-        setSex(value);
-        break;
-      case 'Diagnosis':
-        setDiagnosis(value);
-        break;
-      case 'Assessment tool':
-        setAssessmentTool(value);
-        break;
-      case 'Imaging modality':
-        setImagingModality(value);
-        break;
-      default:
-        break;
-    }
-  }
+  const minAgeBiggerThanMax: boolean = minAge && maxAge ? minAge > maxAge : false;
```

Comment:
neato!

Reply:
how about `minAgeExceedsMaxAge` or adding `Error` in there to drive home that this thing being `true` is a bad thing?